File: sites/cgi-bin/add.py
#!/usr/bin/python

# Import modules for CGI handling
import cgi, cgitb

# Create instance of FieldStorage
form = cgi.FieldStorage()
# Get data from fields

def ft_add(a, b):
    return a + b

if "a" in form and "b" in form:
    a = int(form.getvalue('a'))
    b = int(form.getvalue('b'))

    sum = ft_add(a, b)

    print("<!DOCTYPE html>")
    print("<html lang=\"en\">")
    print("<head>\n")
    print("<meta charset=\"UTF-8\">")
    print("<meta http-equiv=\"X-UA-Compatible\" content=\"IE=edge\">")
    print("<title>Basic Calculator</title>")
    print("</head>")
    print("<body>")
    print("<h1>Result :</h1>")
    print("<p>", a, " + ", b, " = ", sum, ".</p>")
    print("</body>")
    print("</html>")

else:
    print("<!DOCTYPE html>")
    print("<html lang=\"en\">")
    print("<head>\n")
    print("<meta charset=\"UTF-8\">")
    print("<meta http-equiv=\"X-UA-Compatible\" content=\"IE=edge\">")
    print("<title>Basic Calculator</title>")
    print("</head>")
    print("<body>")
    print("<h1>Error :</h1>")
    print("<p>'a' and 'b' parameters required.</p>")
    print("</body>")
    print("</html>")

File: sites/cgi-bin/add.php
<?php

// Check if the request method is POST
if ($_SERVER['REQUEST_METHOD'] === 'POST') {
    $a = $_POST['a'];
    $b = $_POST['b'];
} else {
    // Check if the request method is GET
    $a = $_GET['a'];
    $b = $_GET['b'];
}

// Add the numbers
$result = $a + $b;

// Output the result

echo "<!DOCTYPE html>

<html lang=\"en\">\n
    <head>\n
    <meta charset=\"UTF-8\">\n
    <meta http-equiv=\"X-UA-Compatible\" content=\"IE=edge\">\n
    <title>Basic Calculator</title>\n
</head>\n
<body>";
echo "<h1>Result :</h1>";
echo "<p>$a + $b = $result</p>";
echo "</body>\n
</html>\n";
?>